inference-engine/src/auto_plugin/auto_infer_request.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "auto_infer_request.hpp"
+#include <ie_input_info.hpp>
+#include <cpp_interfaces/interface/ie_iinfer_request_internal.hpp>
+
+namespace AutoPlugin {
+    using namespace InferenceEngine;
+
+AutoInferRequest::AutoInferRequest(const InputsDataMap&   networkInputs,
+                                   const OutputsDataMap&  networkOutputs,
+                                   const InferRequest&    inferRequest)
+    : IInferRequestInternal(networkInputs, networkOutputs)
+    , _inferRequest(inferRequest) {
+    if (_inferRequest) {
```

Comment:
let's leave this for this PR, but pls check if you really need to take the original blobs and store ( then you probably need to update the internal maps when the SetBlob is called). 
I think this is redundant, as the AutoRequest is just a proxy, it has nothing of his own